### ISSUE TITLE
Chip dismiss button has no label

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#4206]
h/t @MaddyDaigle 

## Summary

Right now the button to dismiss a `<Chip />` is `<svg role="button" />` and has no text content. This is causing issues for users of assistive technology not being able to properly interact with it.

## Location 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/7ba3ec98cd8500546314ad90797bc5ed1b1a5fbd/frontend/common/src/components/Chip/Chip.tsx#L60-L70

### Wrap Icon

We should be able to just wrap the `svg` in an `button` with stripped styles and provide some text.

#### Example

```tsx
...
<button 
  onClick={onDismiss} 
  aria-label={intl.formatMessage({
    defaultMessage: "Remove {label}",
    description: "Button text to remove a chip"
  })}>
  <svg />
</button>
...
```

### Wrap Chip

Another option is, we could wrap the entire chip in a button? I'm not sure what appropriate text for this would be but might make it easier increasing the target size for things like:

- Small screens
- Motor control disabilities (Cerebral palsy, parkinsons, etc.)

**Comment by gobyrne** [comment]
@MaddyDaigle can you make the call on this? Consult with other designers if required.

**Comment by gobyrne** [comment]
Pull it into the Sprint as an interruption once decision is made.

**Comment by MaddyDaigle** [comment]
@esizer I agree - the chip should be fully wrapped in a button for motor disabilities. Of course, there should still be an aria-label letting the user know that the skill will be deleted.

I think an aria-label of "Remove {skill}" would be sufficient for screen reader users.
